src/import/trace-event-json.ts

```diff
@@ -0,0 +1,229 @@
+import {FrameInfo} from '../lib/profile'
+import {lastOf} from '../lib/utils'
+import {ProfileBuilderInfo, Sample, StackFrame, TraceEventJsonObject} from './trace-event'
+
+/**
+ * The chrome json trace event spec only specifies name and category
+ * as required stack frame properties
+ *
+ * https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview#heading=h.b4y98p32171
+ */
+function frameInfoForEvent({name, category}: StackFrame): FrameInfo {
+  return {
+    key: `${name}:${category}`,
+    name: name,
+  }
+}
+
+/**
+ * Initialization function to enable O(1) access to the set of active nodes in the stack by node ID.
+ */
+function getActiveNodeArrays(profile: TraceEventJsonObject): Map<number, number[]> {
+  const map: Map<number, number[]> = new Map<number, number[]>()
+
+  // Given a nodeId, `getActiveNodes` gets all the parent nodes in reversed call order
+  const getActiveNodes = (id: number): number[] => {
+    if (map.has(id)) return map.get(id) || []
+
+    const node = profile.stackFrames[id]
+    if (!node) throw new Error(`No such node ${id}`)
+    if (node.parent) {
+      const array = getActiveNodes(node.parent).concat([id])
```

Comment:
This gives me mild pause. This is cloning the array for every child, which seems like it could end up being a ton of memory. If you have a call-stack that's N frames deep, this will allocate O(N^2) memory because the root node will be in N arrays, the one below it in N-1 arrays, etc.

I think this is irrelevant, however, if you switch to using `StackListProfileBuilder` instead though